src/app/project/project.component.ts

```diff
@@ -113,6 +120,160 @@ export class ProjectComponent implements OnInit, OnDestroy {
+      noWrap: true
+    });
+    const World_Imagery = L.tileLayer('https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}', {
+      attribution: 'Tiles &copy; Esri &mdash; Source: Esri, i-cubed, USDA, USGS, AEX, GeoEye, Getmapping, Aerogrid, IGN, IGP, UPR-EGP, and the GIS User Community',
+      maxZoom: 17,
+      noWrap: true
+    });
+
+    this.map = L.map('map', {
+      zoomControl: false, // will be added manually below
+      maxBounds: L.latLngBounds(L.latLng(-90, -180), L.latLng(90, 180)), // restrict view to "the world"
+      zoomSnap: .1 // for greater granularity when fitting bounds
+    });
+
+    // NB: don't need to handle map change events
+    // since we always leave the subject app visible
+
+    // add reset view control
+    this.map.addControl(new resetViewControl());
+
+    // add zoom control
+    L.control.zoom({ position: 'topleft' }).addTo(this.map);
+
+    // add scale control
+    L.control.scale({ position: 'bottomright' }).addTo(this.map);
+
+    // add base maps layers control
+    const baseLayers = {
+      'Ocean Base': Esri_OceanBasemap,
+      'Nat Geo World Map': Esri_NatGeoWorldMap,
+      'World Topographic': World_Topo_Map,
+      'World Imagery': World_Imagery
+    };
+    L.control.layers(baseLayers).addTo(this.map);
+
+    // load base layer
+    for (const key of Object.keys(baseLayers)) {
+      if (key === this.configService.baseLayerName) {
+        this.map.addLayer(baseLayers[key]);
+        break;
+      }
+    }
+
+    // save any future base layer changes
+    this.map.on('baselayerchange', function (e: L.LayersControlEvent) {
+      this.configService.baseLayerName = e.name;
+    }, this);
+
+    // Disable zoom on project details - iterferes with scrolling page.
+    this.map.scrollWheelZoom.disable();
+
+    // draw project marker
+    if (this.project) {
+      const markerIconYellow = L.icon({
+        iconUrl: 'assets/images/marker-icon-yellow.svg',
+        iconSize: [36, 36],
+        iconAnchor: [18, 36],
+        tooltipAnchor: [16, -28]
+      });
+
+      const title = `${this.project.name}\n`
+      + `${this.project.sector}\n`
+      + `${this.project.location}\n`;
+      const marker = L.marker(L.latLng(this.project.centroid[1], this.project.centroid[0]), { title: title })
+        .setIcon(markerIconYellow);
+      this.map.addLayer(marker);
+    }
+    this.map.addLayer(this.appFG);
+
+    this.fixMap();
+  }
+
+  // to avoid timing conflict with animations (resulting in small map tile at top left of page),
+  // ensure map component is visible in the DOM then update it; otherwise wait a bit...
+  // ref: https://github.com/Leaflet/Leaflet/issues/4835
+  // ref: https://stackoverflow.com/questions/19669786/check-if-element-is-visible-in-dom
+  private fixMap() {
+    if (this.elementRef.nativeElement.offsetParent) {
+      this.fitBounds(this.appFG.getBounds());
+
+      // option to centre the map instead of prov view
```

Comment:
I know this was just moved and not written by you, but can we clean up all the commented code while we are in here?